tokio-trace/tokio-trace-core/src/subscriber.rs

```diff
@@ -197,18 +159,16 @@ pub trait Subscriber {
     /// follow from _b_), it may silently do nothing.
     fn add_follows_from(&self, span: &Span, follows: Span);
 
-    // === Filtering methods ==================================================
-
-    /// Returns true if a span with the specified [metadata] would be
-    /// recorded.
+    /// Records that an [`Event`] has occurred.
     ///
-    /// This is used by the dispatcher to avoid allocating for span construction
-    /// if the span would be discarded anyway.
+    /// The provided `Event` struct contains any field values attached to the
+    /// event. The subscriber may pass a [recorder] to the `Event`'s
+    /// [`record` method] to record these values.
     ///
-    /// [metadata]: ::Metadata
-    fn enabled(&self, metadata: &Metadata) -> bool;
-
-    // === Notification methods ===============================================
+    /// [`Event`]: ::event::Event
+    /// [recorder]: ::field::Record
+    /// [`record` method]: ::event::Event::record
+    fn event(&self, event: Event);
```

Comment:
This should probably be passed in by ref?

Reply:
Changed in 07ba2556e45b3c67c2beed08aeb07f8271001b05!